Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7263,1978 @@ def panorama_install_file_content_update_command(args: dict):
+    return result
+
+
+def get_routes(topology: Topology,
+               device_filter_string: str = None) -> ShowRoutingRouteCommandResult:
+    """
+    Pulls all route summary information from the topology
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowRoutingRouteCommandResult = FirewallCommand.get_routes(topology, device_filter_string)
+    return result
+
+
+def get_system_info(topology: Topology,
+                    device_filter_string: str = None) -> ShowSystemInfoCommandResult:
+    """
+    Gets information from all PAN-OS systems in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowSystemInfoCommandResult = UniversalCommand.get_system_info(topology, device_filter_string)
+    return result
+
+
+def get_device_groups(topology: Topology, device_filter_string: str = None) -> List[DeviceGroupInformation]:
+    """
+    Gets the operational information of the device groups in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: List[DeviceGroupInformation] = PanoramaCommand.get_device_groups(topology, device_filter_string)
+    return result
+
+
+def get_template_stacks(topology: Topology, device_filter_string: str = None) -> List[
+    TemplateStackInformation]:
+    """
+    Gets the operational information of the template-stacks in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: List[TemplateStackInformation] = PanoramaCommand.get_template_stacks(topology, device_filter_string)
+    return result
+
+
+def get_topology() -> Topology:
+    """
+    Builds and returns the Topology instance
+    """
+    server_url = demisto.params().get('server')
+    parsed_url = urlparse(server_url)
+    hostname = parsed_url.hostname
+    topology = Topology.build_from_string(
+        hostname,
+        username="",
+        password="",
+        api_key=demisto.params().get('key')
+    )
+    return topology
+
+
+def dataclasses_to_command_results(result: Any, empty_result_message: str = "No results."):
+    """
+    Given a dataclass or list of dataclasses, convert it into a tabular format and finally return CommandResults to demisto.
+
+    :param empty_result_message: If the result data is non
+    """
+    if not result:
+        command_result = CommandResults(
+            readable_output=empty_result_message,
+        )
+        return_results(command_result)
+        return command_result
```

Comment:
why do you use return_results and then return the command_result from the function? This is redundant

Reply:
I suggest to return the command_result from the function and then just call return_results in the main

Reply:
I'm not sure why I did it this way originally - I've removed that.